File: server/models/shop_response.js
const mongoose = require("mongoose");

const shopResponseSchema = mongoose.Schema(
  {
    requestId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "CustomerRequest",
      required: true,
    },
    shopUserId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    shopProfileId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "ShopProfile",
      required: true,
    },
    price: {
      type: String,
      required: true,
      trim: true,
    },
    message: {
      type: String,
      required: true,
      trim: true,
    },
  },
  { timestamps: true }
);

shopResponseSchema.index({ requestId: 1, shopUserId: 1 }, { unique: true });

const ShopResponse = mongoose.model("ShopResponse", shopResponseSchema);

module.exports = ShopResponse;

File: lib/features/requests/domain/entities/shop_response.dart
class ShopResponse {
  const ShopResponse({
    required this.shopId,
    required this.name,
    required this.distance,
    required this.respondedAgo,
    required this.rating,
    required this.reviews,
    required this.message,
    required this.price,
    this.shopUserId = '',
  });

  final String shopId;
  final String shopUserId;
  final String name;
  final String distance;
  final String respondedAgo;
  final String rating;
  final String reviews;
  final String message;
  final String price;
}
